File: Chronos.DotNet.SqlProfiler/source/Chronos.DotNet.SqlProfiler/MsSqlQueryInfo.cs
using System;
using Chronos.Model;

namespace Chronos.DotNet.SqlProfiler
{
    [Serializable]
    public sealed class MsSqlQueryInfo : UnitBase
    {
        public MsSqlQueryInfo(MsSqlQueryNativeInfo queryInfo)
            : base(queryInfo)
        {
            SetDependencies();
        }

        private MsSqlQueryNativeInfo QueryNativeInfo
        {
            get { return (MsSqlQueryNativeInfo)NativeUnit; }
        }

        internal void SetDependencies()
        {
        }
    }
}

File: source/Chronos.DotNet.SqlProfiler/Chronos.DotNet.SqlProfiler.Agent/MsSqlQueryInfo.cpp
#include "stdafx.h"
#include "Chronos.DotNet.SqlProfiler.Agent.h"

namespace Chronos
{
	namespace Agent
	{
		namespace DotNet
		{
			namespace SqlProfiler
			{
				MsSqlQueryInfo::MsSqlQueryInfo()
				{

				}

				void MsSqlQueryInfo::InitializeName(__string* queryText)
				{
					_name = queryText;
				}
			}
		}
	}
}